VDP---Enhanced-Audio-API.md

```diff
@@ -525,7 +588,22 @@ When changing the sample rate for the underlying audio system, the frequency of
 
 (NB there is a bug in the underlying audio system that means that changing the underlying sample rate will cause the pitch of notes currently being played to change.  If the next note played on a channel is at exactly the same frequency as the last note that will also play at an incorrect pitch.  Playing a note at a different frequency will clear this issue and all subsequent notes will be played at the correct pitch.  This issue does not affect sample playback.  This bug will be fixed in a future release.)
 
-This command does not currently return a value.
+Returns 1 on success, 0 for failure.
+
+Prior to Console8 VDP 2.5.0 this command did not return a status.
 
 This command was added in the Console8 VDP 2.2.0 release.
 
+
+## Command 14: Set channel waveform parameters
+
+`VDU 23, 0, &85, channel, 14, parameter, value`
+
+The exact use of this command will vary depending upon the waveform type being used by the channel.
+
+At present only one waveform type, the square wave, supports a parameter, which sets the duty cycle for the square wave.
+
+Returns 1 on success, 0 for failure.
+
+This command was added in the Console8 VDP 2.5.0 release.
```

Comment:
heh - as of right now, this command does not yet exist :grin:

hopefully I'll add it for VDP 2.5.0.  adjusting the duty cycle on the square wave is a missing feature, and its absence means we can't do square-wave sounds in the style of the NES and similar consoles

Reply:
command now exists and has been fleshed out a bit more